contracts/eosio.system/include/eosio.system/eosio.system.hpp

```diff
@@ -509,11 +509,30 @@ namespace eosiosystem {
       uint64_t  primary_key() const { return giftee.value; } // unique as one giftee can only hold gifted ram from one gifter
    };
 
+   struct [[eosio::table("peerkeysver"), eosio::contract("eosio.system")]] peer_keys_version {
+      uint64_t   version;                                  // version incremented every time peer_keys_table is modified
```

Comment:
I think you can get around the need for tracking this (and the need for the table as a whole) by instead storing the block num a row is modified on. It's not a change I really feel strongly about, but it would be more efficient.

Reply:
Good idea, but I thought that, in addition to the version, I would also track the number of deleted keys.

Reply:
ah yes that is true, deletion is more tricky that way

Reply:
> by instead storing the block num a row is modified on

Is there actually a way to get the current block number in a contract? It would be nice to have it to resolve the fork switch issue, but from [this](https://github.com/EOSIO/eos/issues/6695) discussion I wonder if it is available?

Reply:
yep it was added in the `GET_BLOCK_NUM` protocol feature